WeakAuras/Prototypes.lua

```diff
@@ -1126,20 +1126,35 @@ local function AddUnitChangeInternalEvents(unit, t)
   end
 end
 
+local function AddUnitEventForEvents(unit, result, event)
+  if not WeakAuras.baseUnitId[unit] then
+    tinsert(result, event)
+  end
+end
+
+local function AddUnitEventForUnit_events(unit, result, event)
+  if WeakAuras.baseUnitId[unit] then
+    result[unit] = result[unit] or {}
+    tinsert(result[unit], event)
+  end
+end
+
 WeakAuras.event_prototypes = {
   ["Unit Characteristics"] = {
     type = "status",
     events = function(trigger)
-      local result = {
-        "UNIT_LEVEL",
-        "UNIT_FACTION"
-      };
-      AddUnitChangeEvents(trigger.unit, result);
-      if trigger.unitisunit then
-        AddUnitChangeEvents(trigger.unitisunit, result);
-      end
+      local result = {}
+      AddUnitEventForEvents(trigger.unit, result, "UNIT_LEVEL")
+      AddUnitEventForEvents(trigger.unit, result, "UNIT_FACTION")
+      AddUnitChangeEvents(trigger.unit, result)
```

Comment:
I think there is an optimization to do on AddUnitChangeEvents, have to think more about it